src/main/java/net/minecraftforge/common/smelting/SmeltingRecipeRegistry.java

```diff
@@ -0,0 +1,194 @@
+     * with {@code strict} set to false.
+     *
+     * @param input  the input stack
+     * @param output the output stack
+     */
+    public static void addSimpleRecipe(@Nonnull ItemStack input, @Nonnull ItemStack output)
+    {
+        addSimpleRecipe(input, output, DEFAULT_COOK_TIME);
+    }
+
+    /**
+     * Add a simple recipe that smelts the stacks matching the given input stack into the output stack
+     * in the given duration.
+     * Stack matching is done according to {@link OreDictionary#itemMatches(ItemStack, ItemStack, boolean)}
+     * with {@code strict} set to false.
+     *
+     * @param input    the input stack
+     * @param output   the output stack
+     * @param duration the duration for this recipe in ticks
+     */
+    public static void addSimpleRecipe(@Nonnull ItemStack input, @Nonnull ItemStack output, int duration)
+    {
+        addRecipe(new SimpleSmeltingRecipe(input, output, duration));
+    }
+
+    /**
+     * Add a recipe that smelts stacks matching the given OreDictionary name into the output stack.
+     *
+     * @param input  the OreDictionary name
+     * @param output the output stack
+     */
+    public static void addOreRecipe(@Nonnull String input, @Nonnull ItemStack output)
+    {
+        addOreRecipe(input, output, DEFAULT_COOK_TIME);
+    }
+
+    /**
+     * Add a recipe that smelts stacks matching the given OreDictionary name into the output stack
+     * in the given duration.
+     *
+     * @param input    the OreDictionary name
+     * @param output   the output stack
+     * @param duration the duration for this recipe in ticks
+     */
+    public static void addOreRecipe(@Nonnull String input, @Nonnull ItemStack output, int duration)
+    {
+        addRecipe(new OreSmeltingRecipe(input, output, duration));
+    }
+
+    /**
+     * Get the output stack for the given input. The result is always a defensive copy.
+     *
+     * @param input the input stack
+     * @return the output stack
+     */
+    @Nonnull
+    public static ItemStack getOutput(ItemStack input)
+    {
+        SmeltingRecipe recipe = getMatchingRecipe(input);
+        return recipe == null ? ItemStack.field_190927_a : recipe.getOutput(input);
+    }
+
+    /**
+     * Get the duration the given input stack takes to smelt.
+     *
+     * @param input the input stack
+     * @return the duration in ticks
+     */
+    public static int getDuration(ItemStack input)
+    {
+        for (SmeltingRecipe recipe : recipes)
+        {
+            if (recipe.matches(input))
+            {
+                return recipe.getDuration(input);
+            }
+        }
+        return 0;
+    }
+
+    @Deprecated // internal API, do not use!
+    public static void addVanillaRecipe(@Nonnull ItemStack input, @Nonnull ItemStack output)
+    {
+        addRecipe(new VanillaSmeltingRecipe(input, output));
+    }
+
+    @Deprecated // internal API
+    public static void getAllPossibleItems(Collection<Item> target)
```

Comment:
`getAllPossibleOutputs` may be a better name. 
Probably no need to mark this as an internal API, I'm sure someone could find a use for it.